web/src/components/layouts/layout.tsx

```diff
@@ -136,31 +136,43 @@ export default function Layout(props: PropsWithChildren) {
     !router.pathname.startsWith("/public/")
   ) {
     void signOut({
-      callbackUrl: "/auth/sign-in",
+      callbackUrl: `/auth/sign-in`,
     });
     return <Spinner message="Redirecting" />;
   }
 
   if (
     session.status === "unauthenticated" &&
     !unauthenticatedPaths.includes(router.pathname) &&
-    !publishablePaths.includes(router.pathname) &&
     !router.pathname.startsWith("/public/")
   ) {
-    void router.replace("/auth/sign-in");
+    const targetUrl = router.asPath;
+    if (targetUrl && targetUrl !== "/") {
+      void router.replace(
+        `/auth/sign-in?targetUrl=${encodeURIComponent(targetUrl)}`,
+      );
+    } else {
+      void router.replace(`/auth/sign-in`);
+    }
     return <Spinner message="Redirecting" />;
   }
 
   if (
     session.status === "authenticated" &&
     unauthenticatedPaths.includes(router.pathname)
   ) {
-    void router.replace("/");
+    const targetUrl = router.query.targetUrl as string | undefined;
+    if (targetUrl) {
+      void router.replace(targetUrl);
+    } else {
+      void router.replace("/");
+    }
     return <Spinner message="Redirecting" />;
   }
 
   const hideNavigation =
     session.status === "unauthenticated" ||
+    projects.length === 0 ||
```

Comment:
this was removed in a recent change to main, please remove

Reply:
Done.

Reply:
@marcklingen is this okay or does it need more changes? Also I noticed the tests CI tests are failing for forked repositories, is this intended behaviour or do I need to change some config?

Reply:
we fixed the ci test, sorry for the confusion here